docs/rfcs/1008-nightly-build.md

```diff
@@ -0,0 +1,206 @@
+  - [Improve the unique dev format](#improve-the-unique-dev-format)
+- [F.A.Q](#faq)
+  - [Why do we need to release the nightly build? Or why we can't directly use the artifact link of a workflow?](#why-do-we-need-to-release-the-nightly-build-or-why-we-cant-directly-use-the-artifact-link-of-a-workflow)
+  - [Why do we use the `dev` identifier for a nightly build ?](#why-do-we-use-the-dev-identifier-for-a-nightly-build-)
+
+## Goals
+
+- Periodically test the release pipeline.
+- Have a recent build software at the ready (client and server).
+
+## What we already have
+
+We already have a workflow that create a release & build the artifact[^releaser-workflow] for that release.
+
+That workflow[^releaser-workflow] is triggered when we modify the workflow itself but notably when we push a tag on the repository that have the format `v[0-9]+.[0-9].[0-9]+*`.
+
+Currently, the workflow build the server python wheel for Linux, Windows & macOS.
+
+[^releaser-workflow]: https://github.com/Scille/parsec-cloud/blob/05e4deba964c6b68b6fcbca4d458c235c43f6cff/.github/workflows/releaser.yml
+
+The script `releaser.py` can generate unique version that can be used for versioning and tagging nightly build.
+
+## What is missing
+
+From the release, we already have the server part indicated by:
+
+<https://github.com/Scille/parsec-cloud/blob/05e4deba964c6b68b6fcbca4d458c235c43f6cff/.github/workflows/releaser.yml#L74-L79>
+
+### For the client part
+
+#### Missing version argument in the workflow `package-ionic-app.yml`
+
+The workflow ([`package-ionic-app.yml`]) need to be adapted like [`package-server.yml`] to take a version in argument.
+
+[`package-server.yml`]: https://github.com/Scille/parsec-cloud/blob/05e4deba964c6b68b6fcbca4d458c235c43f6cff/.github/workflows/package-server.yml
+
+#### Include the client part in the releaser workflow
+
+We need to add the client to the workflow [`releaser.yml`].
+Thankfully we already have the workflow for that ([`package-ionic-app.yml`]).
+
+[`package-ionic-app.yml`]: https://github.com/Scille/parsec-cloud/blob/05e4deba964c6b68b6fcbca4d458c235c43f6cff/.github/workflows/package-server.yml
+[`releaser.yml`]: https://github.com/Scille/parsec-cloud/blob/05e4deba964c6b68b6fcbca4d458c235c43f6cff/.github/workflows/releaser.yml
+
+### Overwriting nightly release & automatic release
+
+Currently, the [`releaser.yml`] workflow create a [GitHub release] in draft mode.
+
+[GitHub release]: https://docs.github.com/en/repositories/releasing-projects-on-github/about-releases
+
+This is problematic if the workflow create a `draft-release` for each nightly build (everyday so).
+
+> We would be overwhelmed by the amount of draft releases that would accumulate.
+
+To circumvent that we could make the workflow to remove the previous nightly release.
+But we have some limitation, the API to [delete a release] require a `release_id`.
+
+[delete a release]: https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#delete-a-release
```

Comment:
We could do that. I've choose that approach of re-creating the release because GitHub sort the releases by creation date. that mean the nightly release will be put behind older releases.

Reply:
You mean it will be put behind _newer_ releases? That could be inconvenient later, but I think it's no big deal for the moment. Right now, there are no other release planned on the `v2.x` or `v3.x` branch and we need a nightly release _of the upcoming v3.0_. We could improve the nigthly workflow later, based on this RFC.
WDYT?

Reply:
Maybe we can put the two alternatives (update vs delete+create) in the RFC, just to document them so we can merge the PR.

Reply:
For tracking purpose, we have a paragraph that mention `override the assets`